### PR TITLE
Linting fixes, also addresses issue #20

### LINKED ISSUE #20: Changing phone language to Chinese crashes app
Check out time parsing, setting locale to US

### PR DESCRIPTION
**Pull request by kungpaogao**
### Summary

This pull request is applies a lot of fixes to warnings in the lint report (generated using `./gradlew lint`). 

Some fixes include:
- updating dependencies in `build.gradle`
- removing unused resources (sort of, there are still a lot)
- adding more accessibility features (i.e., content descriptions)
- redoing the launcher icons (now adaptive!)
- moving lots of hardcoded text to `string.xml`
- changing left/right to start/end attributes
- changing text size from dp to sp
- general Kotlin fixes (e.g., using Kotlin stdlib for sort, absolute value)
- setting locale for `SimpleDateFormat`, `toUpperCase`, `toLowerCase` to `Locale.US` (fixes #20)

### Screenshots

image

image

### Test Plan

Additional testing is required, but app builds and runs as expected (aside from unintended UI changes mentioned below) on the following devices:

- Huawei Mate 8, API 28 (NXT-AL10)
- Pixel 2 XL, API 27 (emulator)

### Notes 

See below. Additional work is needed to fix these UI changes.

### Breaking Changes 

- updating some of the dependencies (probably material design stuff) resulted in some unintended UI changes, particularly in all selection chips and operating hours section

### CODE REVIEW

File: Density/app/build.gradle [hunk 1]
```diff
@@ -37,24 +37,24 @@ android {
 
 dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
-    api 'com.google.android.material:material:1.1.0-alpha01'
+    api 'com.google.android.material:material:1.1.0-alpha10'
```
Comment (SamChou19815) [inline Density/app/build.gradle]: Yes! Version bumps!

File: Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt [hunk 2]
```diff
@@ -50,11 +48,11 @@ class FacilitiesListAdapter(data: ArrayList<Facility>) : RecyclerView.Adapter<Fa
...
     inner class DescriptionViewHolder(v: View) : FacilitiesListAdapter.ViewHolder(v) {
-        internal var description: TextView
+        internal var description: TextView = v.findViewById(R.id.description_phrase)
 
-        init {
-            description = v.findViewById(R.id.description_phrase)
-        }
+//        init {
+//            description = v.findViewById(R.id.description_phrase)
+//        }
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt]: Remove commented code?

File: Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt [hunk 3]
```diff
@@ -94,19 +92,23 @@ class FacilitiesListAdapter(data: ArrayList<Facility>) : RecyclerView.Adapter<Fa
...
         return object : Filter() {
             override fun performFiltering(charSequence: CharSequence): FilterResults {
-                val charString = charSequence.toString().toLowerCase()
-                if (!charString.isEmpty()) {
-                    val filteredList = ArrayList<Facility>()
-                    for (f in facilities!!) {
-                        if (f.name!!.toLowerCase().contains(charString.toLowerCase())) {
-                            filteredList.add(f)
+                val charString = charSequence.toString().toLowerCase(Locale.US)
+                when(charString.isNotEmpty()) {
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt]: This seems like a misuse of when expression. If the only two cases are `true` and `false`, you should just use if.

File: Density/app/src/main/java/org/cornelldti/density/density/util/APIRequest.java [hunk 4]
```diff
@@ -194,15 +194,15 @@
...
 //                                    if (obj.getString("id")
 //                                            .equals(f_list.get(i).getId())) {
-//                                        f_list.set(i, f_list.get(i).setOccupancy_rating(obj.getInt("density")));
+//                                        f_list.set(i, f_list.get(i).setOccupancyRating(obj.getInt("density")));
 //                                    }
 //                                }
 //                            }
 //
-//                            all_facilities = f_list;
+//                            allFacilities = f_list;
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/util/APIRequest.java]: When can we nuke this all commented out code

File: Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt (hunk 3 above)
Comment (kungpaogao) [reply Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt]: Android Studio was warning me about being able to lift out the if statement?

Comment (SamChou19815) [reply Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt]: I'm thinking about
```kotlin
val charString = charSequence.toString().toLowerCase(Locale.US)
val filterResults = FilterResults()
dataSet = if (charString.isNotEmpty()) {
    val filteredList = ArrayList<Facility>()
    for (f in facilities!!) {
        if (f.name!!.toLowerCase(Locale.US).contains(charString.toLowerCase(Locale.US))) {
            filteredList.add(f)
        }
    }
    Collections.sort(filteredList, SEARCH_SORT.apply(charString))
    filteredList
} else {
    val mutFacilities = facilities!!.toMutableList()
    mutFacilities.sort()
    mutFacilities
}
```

Comment (kungpaogao) [reply Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt]: Should this be lifted out as well? This one is assigned to a value, so I guess changing it to a `when` makes less sense.

Comment (SamChou19815) [reply Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt]: I don't have much context on what's the purpose of database.
In terms of the kotlin code, there exists a way to turn this entire function body into a single expression through extension functions and scope functions (https://kotlinlang.org/docs/reference/scope-functions.html), but I don't think it greatly improves readability. Personally I think the if structure is good enough.

**Comment by SamChou19815** [comment]
@kungpaogao ready to merge?

**Pull request merged**